lib/presentation/features/home/bloc/home_state.dart

```diff
@@ -3,27 +3,31 @@ import 'package:marvelapp_flutter/presentation/models/character_view_data.dart';
 
 class HomeState extends Equatable {
   final bool loading;
+  final bool hasReachedMax;
   final String? error;
   final List<CharacterViewData>? characters;
 
   const HomeState({
     required this.loading,
+    this.hasReachedMax = false,
     this.error,
     this.characters,
```

Comment:
```suggestion
    this.characters = < CharacterViewData >[],
```

Reply:
So how about empty list by default to avoid null checks? Checking isEmpty is more clear.